### PR TITLE
Brings back Pun Pun, soul

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
The bartender's monkey is the single most useless thing in the game. Monkeys in general are useless, but the bar monkey is especially useless because unlike the other, cooler pets, he has no stable identity. This changes that: no longer will the bar monkey just be some randomly named schlub that you dome with a toolbox because you're bored and need to kill five seconds, he is PUN PUN and he deserves RESPECT. 

(He won't get any and will still probably die every round, but at least people might form a tiny bit of attachment to him. Seriously, I've never seen someone refer to the bar monkey by their name, it's always either "bar monkey" or "Pun Pun".)

I left the 10% chance to get a rare name in though, cause "Furious George" and "Dinner" made me laugh. These will still be passed on down the ancestor line as before. Also if he rolls the "Furious George" special name, he starts off PISSED.

## Why It's Good For The Game
Brings soul back to the game, gives Pun Pun back his identity

## Changelog
:cl: Ryll/Shaps
tweak: Pun Pun is Pun Pun again (except in rare cases where he is not)
tweak: Keep an eye out for Furious George though, he's a serious loose cannon
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Not sure how I feel about just yeeting the more common names, but most of them do kinda suck

**Comment by Jared-Fogle** [comment]
They suck, this is good.

File: code/modules/mob/living/carbon/monkey/punpun.dm [hunk 1]
```diff
@@ -1,12 +1,13 @@
 /mob/living/carbon/monkey/punpun //except for a few special persistence features, pun pun is just a normal monkey
 	name = "Pun Pun" //C A N O N
 	unique_name = 0
+	/// If we had one of the rare names in a past life
 	var/ancestor_name
+	/// This does not represent how many rounds Pun Pun has survived, instead it increases each time he dies, and resets when he gibs
```
Comment (Jared-Fogle) [inline code/modules/mob/living/carbon/monkey/punpun.dm:6]:
```suggestion
	/// The number of times Pun Pun has died. Resets when he is gibbed.
```

We don't really need to be documenting what things *aren't*.

**Comment by wesoda25** [comment]
PR of the year

**Comment by cacogen** [comment]
I am glad that at least one thing is sacred. Mr. Dempisi should be a rare name too

**Comment by Tlaltecuhtli** [comment]
who thought giving the bar monkey pope names was a good idea??

**Comment by necromanceranne** [comment]
Soul prs should be instaclosed.

**Comment by wesoda25** [comment]
> Soul prs should be instaclosed.

This isn’t soul, this is pun pun

**Comment by optimumtact** [comment]
make pun pun randomly create drinks using the bar tending machine, at least make him useful, then he might be protected more.

File: code/modules/mob/living/carbon/monkey/punpun.dm (hunk 1 above)
Comment (Ryll-Ryll) [reply code/modules/mob/living/carbon/monkey/punpun.dm:6]: done

**Comment by Ryll-Ryll** [comment]
Paying homage to 40k's Jokaeros and letting monkeys left along with machinery/strange objects make random stuff of greatly varying usefulness sounds like it would make for a pretty funny gimmick, I'll note it for later

File: code/modules/mob/living/carbon/monkey/punpun.dm [hunk 2]
```diff
@@ -3,21 +3,22 @@
...
 	if(ancestor_name)
 		name = ancestor_name
 		if(ancestor_chain > 1)
 			name += " \Roman[ancestor_chain]"
-	else if(prob(5))
-		name = pick(rare_pet_monkey_names)
+	else if(prob(10))
+		name = pick(list("Professor Bobo", "Deempisi's Revenge", "Furious George", "King Louie", "Dr. Zaius", "Jimmy Rustles", "Dinner", "Lanky"))
+		if(name == "Furious George")
+			aggressive = TRUE // Furious George is PISSED
```
Comment (Jared-Fogle) [inline code/modules/mob/living/carbon/monkey/punpun.dm:21]: 👍

**Review by Mothblocks — APPROVED** [review]

**Comment by san7890** [comment]
> make pun pun randomly create drinks using the bar tending machine, at least make him useful, then he might be protected more.

monkeytab

**Pull request merged**